### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Rayan84**
Dear TSE

- We add user prompt for new game, as per previous reviewer's suggestion
- We moved all puts, prints and get.chomps to main.rb file, as per project instructions
- Apply DRY rule

### CODE REVIEW

File: lib/logic.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ def draw_check
   return unless @number_options.size == 1 && @win == ''
 
   @game = false
-  puts '  Draw'
+  output('  Draw')
 end
 
 def win_check
```
Comment (kobiyoyo) [inline lib/logic.rb]: -  [x] Lets separate `win_line` from `win_check` it looks more concise that way

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [x] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [x] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
- [x] concerning this `attr_accessor :name, :mark` its better to  use either attr_readers/attr_writers. Only use attr_accessors if needed and be aware that by using it you are allow writing that property from outside the class. That variable can be set to anything
- [x] Lets validate input for numbers and white space.
## [OPTIONAL]
- [x] We favor the division of code in many methods. (This makes debugging easier and the code more readable). This applies to functions too.
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by ahmetbozaci** [comment]
Dear TSE,

1. According to a review we created two classes in **lib folder.**
- Game: **Game class** include game logic.
- Player: **Player class** include players's names, mark.
    - We deleted **attr_accessor** and use **attr_reader** for mark and name

2. If player writes **a letter with number** ( "1r", "4a") game show to them invalid enter. 
3. If player writes **a number with space** ("1 ", " 5 ") game validate it and update table.
4. We created another method for **win_line variable** and separated it from **win_check** method.

File: lib/player.rb [hunk 2]
```diff
@@ -1,9 +1,8 @@
 # require_relative 'main.rb'
```
Comment (krys2fa) [inline lib/player.rb]: - [x] [REQUIRED] Please remove all comments from your codebase. 
Debugging instructions are not allowed in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)

**Review by krys2fa — CHANGES_REQUESTED** [review]
## Nicely done @ahmetbozaci and @Rayan84  👏 
You did great implementing the requested changes. 💪 

Almost there! Please check below and in the comments for a few more:
### Status: Changes Required ♻️ 

- [x] [REQUIRED] You need to validate all input for player names. Currently, the game accepts empty spaces and numbers as player names. Let's work on this.

Please request for another review after implementing the changes.

Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

File: lib/player.rb (hunk 2 above)
Comment (tirthajyoti-ghosh) [inline lib/player.rb]: - [x] Please remove this comment as pointed out by the previous reviewer.

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    @active_player = player1
+    @win = ''
+    @mark1 = mark1
+    @mark2 = mark2
+  end
+
+  def create_table
+    system('cls')
+    system('clear')
+    puts @table = "
```
Comment (tirthajyoti-ghosh) [inline lib/game.rb]: No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [x] Please remove all `puts/print` statement from this file.

**IMPORTANT!** This is just one place where I am pointing this out. Please go through this file carefully and remove all `puts/print` statements.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## Impressive progress so far 💪! Thanks for making all the changes 👏.

### However, we need to work on some changes before we can merge.

- There are some `puts/print` statements in `/lib/game.rb` file. I have pointed it out in one place. Please go through that file and remove all of those statements.

- There is a commented code on line 1 in `/lib/player.rb`. Please remove this as well.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Ahmet Bozacı & Rayan Rassam** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Remove the .gitkeep file in the lib folder
- [x] Fix rubocop offenses
- [x] The use of "prints" outside the main.rb file is not allowed
- [ ] In the Game class, attributes player1 and player2 are never called outside the class so they don't need the attr_reader 

Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**